components/registry-facade/pkg/registry/registry.go

```diff
@@ -253,51 +254,34 @@ func NewRegistry(cfg config.Config, newResolver ResolverProvider, reg prometheus
 }
 
 func getRedisClient(cfg *config.RedisCacheConfig) (*redis.Client, error) {
-	if cfg.SingleHostAddress != "" {
-		log.WithField("addr", cfg.SingleHostAddress).WithField("username", cfg.Username).Info("connecting to single Redis host")
-		rdc := redis.NewClient(&redis.Options{
-			Addr:     cfg.SingleHostAddress,
-			Username: cfg.Username,
-			Password: cfg.Password,
-		})
-
-		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
-		defer cancel()
-
-		_, err := rdc.Ping(ctx).Result()
-		if err != nil {
-			return nil, xerrors.Errorf("cannot check Redis connection: %w", err)
-		}
-
-		return rdc, nil
+	if cfg.SingleHostAddress == "" {
+		return nil, xerrors.Errorf("registry-facade setting 'singleHostAddr' is missing")
 	}
 
-	if cfg.MasterName == "" {
-		return nil, fmt.Errorf("redis masterName must not be empty")
+	opts := &redis.Options{
+		Addr: cfg.SingleHostAddress,
 	}
-	if len(cfg.SentinelAddrs) == 0 {
-		return nil, fmt.Errorf("redis sentinelAddrs must not be empty")
+
+	if cfg.Username != "" {
+		opts.Username = cfg.Username
 	}
 
-	rdc := redis.NewFailoverClient(&redis.FailoverOptions{
-		MasterName:    cfg.MasterName,
-		SentinelAddrs: cfg.SentinelAddrs,
-		Username:      cfg.Username,
-		Password:      cfg.Password,
+	if cfg.Password != "" {
+		opts.Password = cfg.Password
+	}
 
-		SentinelUsername: cfg.Username,
-		SentinelPassword: cfg.Password,
+	if cfg.UseTLS {
+		opts.TLSConfig = &tls.Config{
+			InsecureSkipVerify: true,
```

Comment:
UseTLS configuration indicates to use insecureSkipVerify is a bit odd to me.
I realized `UseTLS` verifies the certificate, but it seems no.

Reply:
Good point. Refactored to support custom CA if present in the password secret